wire/msgtx.go

```diff
@@ -300,6 +360,33 @@ func (msg *MsgTx) BtcDecode(r io.Reader, pver uint32) error {
 		msg.TxOut[i] = &to
 	}
 
+	// If the transaction's flag byte isn't 0x00 at this point, then one
+	// or more of its inputs has accompanying witness data.
+	if flag[0] != 0 && pver == WitnessVersion {
+		for _, txin := range msg.TxIn {
+			// For each input, the witness is encoded as a stack
+			// with one or more items. Therefore, we first read a
+			// varint which encodes the number of stack items.
+			witCount, err := ReadVarInt(r, pver)
+			if err != nil {
+				return err
+			}
+
+			// Then for witCount number of stack items, each item
+			// has a varint length prefix, followed by the witness
+			// item itself.
+			txin.Witness = make([][]byte, witCount)
```

Comment:
The `witCount` needs to be checked against some type of sane maximum based on the max possible that could fit in a transaction.  Otherwise it would be trivial for an attacker to craft a message with a huge varint encoded here to force a massive allocation of memory and crash the process.